src/main/java/org/folio/fqm/resource/FqlQueryController.java

```diff
@@ -42,7 +42,17 @@ public ResponseEntity<QueryDetails> getQuery(UUID queryId, Boolean includeResult
   @Override
   public ResponseEntity<List<Map<String, Object>>> getContents(ContentsRequest contentsRequest) {
     return ResponseEntity.ok(queryManagementService.getContents(contentsRequest.getEntityTypeId(),
-      contentsRequest.getFields(), contentsRequest.getIds(), Boolean.TRUE.equals(contentsRequest.getLocalize())));
+      contentsRequest.getFields(), contentsRequest.getIds(), null, Boolean.TRUE.equals(contentsRequest.getLocalize()), false));
+  }
+
+  @EntityTypePermissionsRequired(ContentsRequest.class)
```

Comment:
I have no idea how this permission checking system works, and I wrote the thing :D